src/System.Text.Encodings.Web.Utf8/UrlEncoder.cs

```diff
@@ -1,19 +1,336 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Threading.Tasks;
-
-namespace System.Text.Encodings.Web.Utf8
+﻿namespace System.Text.Encodings.Web.Utf8
 {
     public class UrlEncoder
     {
-        public static void UnescapeInPlace(Span<byte> data)
+        /// <summary>
+        /// Unescape a URL path in place.
+        /// </summary>
+        /// <param name="data">The byte span represents a UTF8 encoding url path.</param>
+        /// <returns>
+        /// The length of the unescape sequence on the input byte span. The sequence begins
+        /// from the head of original byte span.
+        /// </returns>
+        /// <remarks>
+        /// The unescape is done in place, which means the result span is the subset of 
+        /// the input span. The beginnings of the spans are the same.
+        /// </remarks>
+        public static int UnescapeInPlace(Span<byte> data)
```

Comment:
I think we should call these Decode/Encode. It's consistent with other APIs we already shipped and with some terminology used even within this file.